modules/db/src/main/scala/gem/dao/StepDao.scala

```diff
@@ -542,5 +542,112 @@ object StepDao {
             ${g.fpu.flatMap(_.toOption)})
         """.update
     }
+
+    object Gnirs {
+      import GnirsConfig.GnirsFpu
+
+      @SuppressWarnings(Array("org.wartremover.warts.Throw"))
+      final case class GnirsFpuBuilder(slit: Option[GnirsFpuSlit], other: Option[GnirsFpuOther]) {
+        def toFpu: GnirsFpu = (slit, other) match {
+          case (Some(s), None) => GnirsFpu.Slit(s)
+          case (None, Some(o)) => GnirsFpu.Other(o)
+          case _               => throw new IllegalArgumentException(s"GNIRS FPU must be either Slit or Other")
+        }
+      }
+
+      final case class GnirsBuilder(
+        camera:               GnirsCamera,
+        decker:               GnirsDecker,
+        disperser:            GnirsDisperser,
+        disperserOrder:       GnirsDisperserOrder,
+        exposureTime:         Duration,
+        filter:               GnirsFilter,
+        prism:                GnirsPrism,
+        fpuBuilder:           GnirsFpuBuilder,
+        readMode:             GnirsReadMode
+      ) {
+        def toGnirs: DynamicConfig.Gnirs =
+          DynamicConfig.Gnirs(
+            camera,
+            decker,
+            disperser,
+            disperserOrder,
+            exposureTime,
+            filter,
+            fpuBuilder.toFpu,
+            prism,
+            readMode
+          )
+      }
```

Comment:
I think you can get rid of the builder altogether, though I'm not sure about the FPU part.  Maybe if it is just a type alias for `Either`.